### DISCUSSION TITLE
Build and test individual app/libs independently

### DISCUSSION THREAD

**Discussion by gilri** [nasa/cFS#538] [Q&A] [answered]
Hello folks, my team and I are relatively new to using cfs. Is there a way to develop a lib that depends on cFS and can later be linked to a cFS app without recompiling the lib?

**Comment by skliper** [comment, marked as answer]
Depending on what you mean by "linked to a cFS app", maybe?

Note cFS libs and apps aren't actually "linked" in the classical compiler linking sense.  You can build libs and apps separately, and they get loaded dynamically separately by ES either at startup as elements from the startup script, or an app can be loaded later via command (note libs can only be loaded at startup and can't be unloaded).

To test and develop separately you'll still need the lib API's defined in a header the app can include and you'll likely want to implement the unit test stubs for the lib to work with.  After that it could be fairly independent.
